frontend/src/components/ProfileDropdown.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { useNavigate } from "react-router-dom";
+import { Routes, Route, BrowserRouter as Router } from "react-router-dom";
+import { useAuth0 } from "@auth0/auth0-react";
+import {
+  Menu,
+  MenuButton,
+  Avatar,
+  MenuList,
+  MenuItem,
+  Button,
+} from '@chakra-ui/react';
+
+interface ProfileDropdownProps {
+  onSignOut: () => void;
+}
+
+const ProfileDropdown: React.FC<ProfileDropdownProps> = ({onSignOut}) => {
+  const { user, isAuthenticated } = useAuth0();
+
+  const navigate = useNavigate();
+
+  if (!isAuthenticated) {
+    return null;
+  }
+
+  const goToProfile = () => {
```

Comment:
Convention is to prefix function names with `handle....`

Reply:
good to know. Implemented